shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -155,21 +180,21 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
       }
       if (!n.name.head.isUpper) err(
         msg"Type names must start with a capital letter", n.toLoc)
-      val td1 = TypeDef(td.kind, td.nme, td.tparams, td.body, baseClassesOf(td), td.toLoc)
+      val td1 = TypeDef(td.kind, td.nme, td.tparams, td.body, td.mthDecls, td.mthDefs, 
+        baseClassesOf(td), td.toLoc)
       allDefs += n.name -> td1
       td1
     }
     import ctx.{tyDefs => oldDefs}
     // implicit val newCtx = ctx.copy(tyDefs = newDefs)
-    ctx.copy(tyDefs = allDefs) |> { implicit ctx =>
+    ctx.copy(env = allEnv, tyDefs = allDefs, targsMaps = allTargsMaps) |> { implicit ctx =>
       ctx.copy(tyDefs = oldDefs ++ newDefs.flatMap { td =>
         implicit val prov: TypeProvenance = tp(td.toLoc, "data definition")
         val n = td.nme
-        val dummyTargs =
-          td.tparams.map(p => freshVar(noProv/*TODO*/)(ctx.lvl + 1))
-        val targsMap: Map[Str, SimpleType] = td.tparams.map(_.name).zip(dummyTargs).toMap
-        val body_ty = typeType(td.body, simplify = false)(ctx.nextLevel, raise, targsMap)
-        var fields = SortedMap.empty[Str, SimpleType]
+        val dummyTargs = td.tparams.map(p => freshVar(noProv/*TODO*/)(ctx.lvl + 1))
+        allTargsMaps += n.name -> MutMap("" -> td.tparams.map(_.name).zip(dummyTargs).toMap)
```

Comment:
There seems to be something unholy going on with type variables here... can you document it? I guess it's for storing mappings for class parameters, associated with no methods. So if it's indeed used as a sentinel value, this would be better expressed with an option type.

Reply:
Addressed in 193deba9ab7b3358683405b91fdcb2f2f4dc1c54